editor/plugins/gdextension_export_plugin.h

```diff
@@ -102,6 +106,82 @@ void GDExtensionExportPlugin::_export_file(const String &p_path, const String &p
 			break;
 		}
 	}
+
+	// Second, try to autodetect
+	if (!could_export && !autodetect_library_prefix.is_empty()) {
+		String autodetect_path = autodetect_library_prefix;
+		if (autodetect_path.is_relative_path()) {
+			autodetect_path = p_path.get_base_dir().path_join(autodetect_path);
+		}
+
+		// Find the folder and file parts of the prefix.
+		String folder;
+		String file_prefix;
+		if (DirAccess::dir_exists_absolute(folder)) {
+			folder = autodetect_path;
+		} else if (DirAccess::dir_exists_absolute(folder.get_base_dir())) {
+			folder = autodetect_path.get_base_dir();
+			file_prefix = autodetect_path.get_file();
+		} else {
+			ERR_FAIL_MSG(vformat("Couldn't export extension: %s. Could not find folder for automatic detection of libraries files. autodetect_library_prefix=\"%s\"", p_path, autodetect_library_prefix));
+		}
+
+		// Open the folder.
+		Ref<DirAccess> dir = DirAccess::open(folder);
+		ERR_FAIL_COND_MSG(!dir.is_valid(), vformat("Couldn't export extension: %s. Could not open folder for automatic detection of libraries files. autodetect_library_prefix=\"%s\"", p_path, autodetect_library_prefix));
+
+		// Iterate the files and check the prefixes, finding the best matching file.
+		Vector<String> best_file_tags;
+		String best_file;
+		dir->list_dir_begin();
+		String file_name = dir->_get_next();
+		while (file_name != "") {
+			if (!dir->current_is_dir() && file_name.begins_with(file_prefix)) {
+				Vector<String> tags = file_name.trim_prefix(file_prefix).split(".");
+				Vector<String> file_tags;
+				for (int i = 0; i < tags.size(); i++) {
+					String tag = tags[i].strip_edges();
+					if (p_features.has(tag)) {
+						file_tags.append(tag);
+					}
+				}
+				if (file_tags.size() > best_file_tags.size()) {
```

Comment:
It should unconditionally reject files with any tag that is not in `p_features`.

Or something like `libname.wrong_os.release.standalone.etc.etc2.wrong_arc` will have higher priority (4 matching tags: `release`, `standalone`, `etc`, `etc2`) than `libname.correct_os.release.correct_arch` (3 matching tags).

Reply:
The problem is identifying parts of the filename that are tags and those that are not. 
I suppose a solution might be to simply consider that everything after the prefixes are tags besides the extension, but that mean users won't be able to have any custom content after the prefix.

Reply:
Restricting a file name to allow only tags and extension after the prefix seems reasonable. Or we can have a full list of supported tags (all archs and platforms).

Reply:
Ah yeah, with a full list of arch and platforms it could work, but I wonder if users can define their one features. In such case I am not sure it would be possible, unless those features can be listed and transformed into identifiable tags 🤔